### PR TITLE
feat: add endpoints

### PR DESCRIPTION
**Pull request by ai-n0808**
Added these endpoints.

<Places>
- GET /places : Retrieves the Place interfaced object, no just ‘Places table rows’
- POST /places : Add a new place to the database
- PATCH /places/:id : Update the details of a specific place.
- DELETE /places/:id : Remove a place by its ID.

<Experiences>
- POST /places/:id/experiences : Add a new experience for a specific place.
- PATCH /places/:id/experiences/:expId Update details of a specific experience by its ID.
- DELETE /places/:id/experiences/:expId : Remove an experience by its ID.

### CODE REVIEW

**Comment by ai-n0808** [comment]
@brianaqp 
Please let me know if you have any thought or feedback for this.

**Comment by ai-n0808** [comment]
Now If I test endpoints in postman, I got error with endpoints of experiences. 
All of places endpoints are all good.

File: src/router/index.ts [hunk 1]
```diff
@@ -1,16 +1,122 @@
...
 router.get("/places", async (req: Request, res: Response) => {
-  const data = GeneralModel.getInitialData();
+  try {
+    const places = await prisma.places.findMany({
+      include: {
+        experiences: true,
+        images: true,
+      },
+    });
+    res.json(places);
```
Comment (brianaqp) [inline src/router/index.ts:19]: For know its ok, it doesn't have the Place Interface structure, but its very close

File: src/router/index.ts [hunk 2]
```diff
@@ -1,16 +1,122 @@
...
+  } catch (error) {
+    res.status(500).json({ error });
+  }
+});
+
+//Add a new place
+router.post("/places", async (req: Request, res: Response) => {
```
Comment (brianaqp) [inline src/router/index.ts:27]: We are not going to use this, most likely in the entire project

When we add the feature of "add your own place" we can implemented, but for now, removed

Places are seed it manually, or by admin

File: src/router/index.ts [hunk 3]
```diff
@@ -1,16 +1,122 @@
...
+    const newPlace = await prisma.places.create({
+      data: placeData,
+    });
+    res.status(201).json(newPlace);
+  } catch (error) {
+    res.status(400).json({ error });
+  }
+});
+
+//Update details of specific place
```
Comment (brianaqp) [inline src/router/index.ts:40]: The same, we are not gonna used this

File: src/router/index.ts [hunk 4]
```diff
@@ -1,16 +1,122 @@
...
+router.patch("/places/:id", async (req: Request, res: Response) => {
```
Comment (brianaqp) [inline src/router/index.ts:41]: Delete it

File: src/router/index.ts [hunk 5]
```diff
@@ -1,16 +1,122 @@
...
+    console.error(error);
+    res.status(400).json({ error });
+  }
+});
 
-  res.send({
-    data,
-  });
+//Remove place by its ID
+router.delete("/places/:id", async (req: Request, res: Response) => {
```
Comment (brianaqp) [inline src/router/index.ts:57]: Delete

File: src/router/index.ts [hunk 6]
```diff
@@ -1,16 +1,122 @@
...
+    res.status(204).send();
+  } catch (error) {
+    res.status(400).json({ error });
+  }
 });
 
+// --! Experiences endpoints
+
+//Retrieve all experiences related to a specific place
+router.post("/places/:id/experiences", async (req: Request, res: Response) => {
```
Comment (brianaqp) [inline src/router/index.ts:73]: Looks nice. 👍 we need to test it later

File: src/router/index.ts [hunk 7]
```diff
@@ -1,16 +1,122 @@
...
+        where: { id: Number(expID) },
+        data: updateData,
+      });
+      res.json(updatedExperience);
+    } catch (error) {
+      console.error(error);
+      res.status(400).json({ error });
+    }
+  }
+);
```
Comment (brianaqp) [inline src/router/index.ts:108]: Nice +1

File: src/router/index.ts [hunk 8]
```diff
@@ -1,16 +1,122 @@
...
+      await prisma.experiences.delete({
+        where: { id: Number(expID) },
+      });
+      res.status(204).send();
+    } catch (error) {
+      res.status(400).json({ error });
+    }
+  }
+);
+
```
Comment (brianaqp) [inline src/router/index.ts:125]: Nice +1

**Review by brianaqp — CHANGES_REQUESTED** [review]
Everything looks fine, perfect. Just delete the unused endpoints, and later we test

**Review by brianaqp — APPROVED** [review]

**Pull request merged**